src/Swashbuckle.AspNetCore/Swashbuckle.AspNetCore.csproj

```diff
@@ -1,23 +1,50 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
-
+﻿<Project>
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
   <Import Project="..\..\version.props" />
 
   <PropertyGroup>
     <Description>Swagger tools for documenting APIs built on ASP.NET Core</Description>
     <TargetFramework>netstandard2.0</TargetFramework>
-    <AssemblyName>Swashbuckle.AspNetCore</AssemblyName>
     <PackageId>Swashbuckle.AspNetCore</PackageId>
     <PackageTags>swagger;documentation;discovery;help;webapi;aspnet;aspnetcore</PackageTags>
     <PackageProjectUrl>https://github.com/domaindrivendev/Swashbuckle.AspNetCore</PackageProjectUrl>
     <PackageLicenseUrl>https://raw.githubusercontent.com/domaindrivendev/Swashbuckle.AspNetCore/master/LICENSE</PackageLicenseUrl>
     <RepositoryType>git</RepositoryType>
     <RepositoryUrl>https://github.com/domaindrivendev/Swashbuckle.AspNetCore.git</RepositoryUrl>
+
+    <!-- Package contains only the final .nuspec file with a few listed dependencies. -->
+    <GenerateNuspecDependsOn>$(GenerateNuspecDependsOn);PopulateNuspec</GenerateNuspecDependsOn>
+    <IncludeBuildOutput>false</IncludeBuildOutput>
+    <IncludeContentInPack>false</IncludeContentInPack>
+    <IncludeSource>false</IncludeSource>
+    <IncludeSymbols>false</IncludeSymbols>
+    <MicrosoftExtensionsApiDescriptionServerPackageVersion>0.3.0-preview6.19307.2</MicrosoftExtensionsApiDescriptionServerPackageVersion>
```

Comment:
@dougbu Now that it's out of preview, I'm looking to upgrade the version of `ApiDescription.Server` that comes bundled with Swashbuckle. This brought me back to this PR, and now that I'm looking at it a little closer, I have a couple of questions I was hoping you could answer, so that I can understand things a little better:

1. Why does `Swashbuckle` need the dependency on the `ApiDescription.Server` package? It's not referenced anywhere in the SB code. If the goal is to introduce additional tooling, wouldn't it be more appropriate for the application to take that dependency instead of it being bundled in with SB?

2. In this PR, what was the rationale behind introducing an explicit .nuspec file and target to populate it? I preferred the simplicity (e.g. less moving parts) when it was just a single .csproj file. IIRC that was working just fine - was there something you needed for the `ApiDescription.Server` stuff that wouldn't have worked with the original approach? If so, can you let me know what that was? Thanks

Reply:
@domaindrivendev 
1. Swashbuckle needs a dependency on ApiDescription.Server to reduce the friction for those who want document generation on build. As we've discussed offline, it makes sense for this feature to be opt-in.
  Opting in should be as simple as possible and we made a choice a while ago to hide the Microsoft.Extensions.ApiDescription.Server package (which is very small) from users. It's completely unusable without Swashbuckle.AspNetCore in any case.
2. I had to use an explicit .nuspec because NuGet doesn't handle the case of a package that depends on a dev-only package without help. To avoid making this too horrendous, I pass properties from the .csproj for everything -- trying to avoid a need to edit the .nuspec in the future (unless the package content needs to change).
  In any case, making the doc generation features opt-in needs a tiny (single property) Swashbuckle.AspNetCore.props file in your package. The .nuspec makes this change trivial.

/cc @bradygaster

Reply:
(Generally, packages containing MSBuild goop are difficult to manage w/o .nuspec files.)